Libraries/LibPCIDB/Database.cpp

```diff
@@ -0,0 +1,205 @@
+#include <AK/RefPtr.h>
+#include <AK/StringView.h>
+
+#include "Database.h"
+
+namespace PCIDB {
+
+RefPtr<Vendor> Database::get_vendor(u16 vendor_id) const
+{
+    auto res = m_vendors.get(vendor_id);
+    if (res.has_value())
+        return res.value();
+    return {};
+}
+
+RefPtr<Device> Database::get_device(u16 vendor_id, u16 device_id) const
+{
+    auto parent = get_vendor(vendor_id);
+    if (!parent)
+        return {};
+
+    auto res = parent->devices.get(device_id);
+    if (res.has_value())
+        return res.value();
+    return {};
+}
+
+RefPtr<Subsystem> Database::get_subsystem(u16 vendor_id, u16 device_id, u16 subvendor_id, u16 subdevice_id) const
+{
+    auto parent = get_device(vendor_id, device_id);
+    if (!parent)
+        return {};
+
+    auto res = parent->subsystems.get((subvendor_id << 16) + subdevice_id);
+    if (res.has_value())
+        return res.value();
+    return {};
+}
+
+RefPtr<Class> Database::get_class(u8 class_id) const
+{
+    auto res = m_classes.get(class_id);
+    if (res.has_value())
+        return res.value();
+    return {};
+}
+
+RefPtr<Subclass> Database::get_subclass(u8 class_id, u8 subclass_id) const
+{
+    auto parent = get_class(class_id);
+    if (!parent)
+        return {};
+
+    auto res = parent->subclasses.get(subclass_id);
+    if (res.has_value())
+        return res.value();
+    return {};
+}
+
+RefPtr<ProgrammingInterface> Database::get_programming_interface(u8 class_id, u8 subclass_id, u8 programming_interface_id) const
+{
+    auto parent = get_subclass(class_id, subclass_id);
+    if (!parent)
+        return {};
+
+    auto res = parent->programming_interfaces.get(programming_interface_id);
+    if (res.has_value())
+        return res.value();
+    return {};
+}
+
+u8 parse_hex_digit(char digit)
+{
+    if (digit >= '0' && digit <= '9')
+        return digit - '0';
+    ASSERT(digit >= 'a' && digit <= 'f');
+    return 10 + (digit - 'a');
+}
+
+template<typename T>
+T parse_hex(StringView str, int count)
+{
+    ASSERT(str.length() >= count);
+
+    T res = 0;
+    for (int i = 0; i < count; i++)
+        res = (res << 4) + parse_hex_digit(str[i]);
+
+    return res;
+}
+
+int Database::init()
+{
+    if (m_ready)
+        return 0;
+
+    if (!m_file.is_valid())
+        return -1;
+
+    m_view = StringView((char*)m_file.pointer(), m_file.size());
```

Comment:
We don't want to look like we intend to mutate the contents, so:
```suggestion
    m_view = StringView((const char*)m_file.pointer(), m_file.size());
```